iree/base/internal/dynamic_library_posix.c

```diff
@@ -31,35 +31,61 @@ struct iree_dynamic_library_t {
   void* handle;
 };
 
-// Returns true if the temp files we extract should be kept after the library
-// is closed. This is useful for tooling such as perf that need to find the
-// shared libraries after the program has exited or atexit tools like tracy that
-// require the library still be loaded.
-//
-// To override:
-//   IREE_PRESERVE_DYLIB_TEMP_FILES=1 iree-run-module ...
-static bool iree_dynamic_library_should_preserve_temp_files() {
-  return getenv("IREE_PRESERVE_DYLIB_TEMP_FILES") != NULL;
+// Holds information about where to store temporary dynamic library files, and
+// whether to preserve them or delete them.
+typedef struct {
+  const char* temp_dir;
+  bool preserve_temp_files;
+} iree_dynamic_library_temp_files_params_t;
+
+static iree_status_t iree_dynamic_library_get_temp_files_params(
```

Comment:
We can use iree_call_once to do this check once - as-is we'll be doing the env checks, string compares, and the filesystem stat each time we load a library, even though all that information never changes. 

you can see how this is used in the win32 loader to do this same thing:
https://github.com/google/iree/blob/931a3b1465246e81ea99d8d7b308626c50709238/iree/base/internal/dynamic_library_win32.c#L88

https://github.com/google/iree/blob/931a3b1465246e81ea99d8d7b308626c50709238/iree/base/internal/dynamic_library_win32.c#L59-L62

Unlike in the win32 case where it needs a copy of the path string here your struct is fine as env memory and your string literals are read-only.

Reply:
This is a preexisting issue and I had also noticed it once working on this code. Filed Issue #8656 about it --- good first issue, too. I can do it now if you feel that's important, but as this is preexisting and the present PR is part of a deep chain (see #8655) that's on the critical path for a bunch of immediate work, please consider not requiring this synchronously.

Reply:
@benvanik let me know if that's OK to defer?

Reply:
The previous code didn't make a syscall - I really would prefer to avoid sloppy syscalls as it makes strace and such harder to parse and it's the kind of issue that is never going to get fixed. Just copy/paste the iree_call_once call - it'd have been less typing than the issue or these comments :)

Reply:
OK, done! please take another look. It wasn't 100% trivial: how would i return a `iree_status_t` from static initialization? Given the potential heap allocation in a `iree_status_t`, it didn't feel comfortable to have it as a global. so i just made that a `bool`, and the caller has the info to generate the `iree_status_t` from that.

Reply:
thanks!